src/v/cloud_storage/tests/topic_manifest_test.cc

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/vectorizedio/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "bytes/iobuf.h"
+#include "bytes/iobuf_parser.h"
+#include "cloud_storage/topic_manifest.h"
+#include "cloud_storage/types.h"
+#include "cluster/types.h"
+#include "model/metadata.h"
+#include "seastarx.h"
+
+#include <seastar/testing/test_case.hh>
+#include <seastar/testing/thread_test_case.hh>
+
+#include <boost/test/tools/old/interface.hpp>
+#include <boost/test/unit_test.hpp>
+
+using namespace cloud_storage;
+
+// update manifest, serialize, compare jsons
+
+static const cluster::topic_configuration cfg(
+  model::ns("cfg-test-namespace"),
+  model::topic("cfg-test-topic"),
+  /* partition_count = */ 32,
+  /* replication_factor = */ 3);
+
+static constexpr std::string_view min_topic_manifest_json = R"json({
+    "version": 1,
+    "namespace": "test-namespace",
+    "topic": "test-topic",
+    "partition_count": 32,
+    "replication_factor": 3,
+    "revision_id": 0,
+    "compression": null,
+    "cleanup_policy_bitflags": null,
+    "compaction_strategy": null,
+    "timestamp_type": null,
+    "segment_size": null
+})json";
+
+static constexpr std::string_view full_topic_manifest_json = R"json({
+    "version": 1,
+    "namespace": "full-test-namespace",
+    "topic": "full-test-topic",
+    "partition_count": 64,
+    "replication_factor": 6,
```

Comment:
nit: we don't support even replication factors :)

Reply:
but the code parses it, so let's fix with behavior in tests